assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Footer.js

```diff
@@ -47,116 +47,110 @@ import SourceLink from '../../../../../components/SourceLink';
-		select( MODULES_SEARCH_CONSOLE ).isDomainProperty()
-	);
-	const referenceSiteURL = useSelect( ( select ) =>
-		untrailingslashit( select( CORE_SITE ).getReferenceSiteURL() )
-	);
-	const url = useSelect( ( select ) =>
-		select( CORE_SITE ).getCurrentEntityURL()
-	);
-	const dateRangeDates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
-			offsetDays: DATE_RANGE_OFFSET,
-		} )
-	);
+function SourceLinkAnalytics( { id } ) {
+	const serviceURL = useSelect(
+		( select ) => {
+			const { getServiceReportURL } = select( MODULES_ANALYTICS );
+			const url = select( CORE_SITE ).getCurrentEntityURL();
+			const rangeDates = select( CORE_USER ).getDateRangeDates( {
+				compare: true,
+				offsetDays: DATE_RANGE_OFFSET_ANALYTICS,
+			} );
+			const drilldownArgs = isURL( url )
+				? [ `analytics.pagePath:${ getURLPath( url ) }` ]
+				: [];
 
-	const searchConsoleDeepLinkArgs = {
-		resource_id: propertyID,
-		metrics: metrics[ selectedStats ]?.metric,
-		...generateDateRangeArgs( dateRangeDates ),
-	};
-	if ( url ) {
-		searchConsoleDeepLinkArgs.page = `!${ url }`;
-	} else if ( isDomainProperty && referenceSiteURL ) {
-		searchConsoleDeepLinkArgs.page = `*${ referenceSiteURL }`;
-	}
-	const searchConsoleDeepLink = useSelect( ( select ) =>
-		select( MODULES_SEARCH_CONSOLE ).getServiceReportURL(
-			searchConsoleDeepLinkArgs
-		)
-	);
+			switch ( id ) {
+				case 'users':
+					return getServiceReportURL( 'acquisition-channels', {
+						...generateAnalyticsDateRangeArgs( rangeDates ),
+						'_r.drilldown': [
+							...drilldownArgs,
+							'analytics.trafficChannel:Organic Search',
+						].join( ',' ),
+					} );
 
-	const analyticsDrilldownArgs = [];
-	if ( isURL( url ) ) {
-		analyticsDrilldownArgs.push(
-			`analytics.pagePath:${ getURLPath( url ) }`
-		);
-	}
-	const analyticsRangeDates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
-			compare: true,
-			offsetDays: DATE_RANGE_OFFSET_ANALYTICS,
-		} )
-	);
-	const analyticsGoalsDeepLink = useSelect( ( select ) =>
-		select( MODULES_ANALYTICS ).getServiceReportURL(
-			'conversions-goals-overview',
-			{
-				...generateAnalyticsDateRangeArgs( analyticsRangeDates ),
-				'_r.drilldown': analyticsDrilldownArgs.join( ',' ),
-			}
-		)
-	);
+				case 'goals':
+					return getServiceReportURL( 'conversions-goals-overview', {
+						...generateAnalyticsDateRangeArgs( rangeDates ),
+						'_r.drilldown': drilldownArgs.join( ',' ),
+					} );
 
-	const analyticsVisitorsOverviewDeepLink = useSelect( ( select ) =>
-		select( MODULES_ANALYTICS ).getServiceReportURL( 'visitors-overview', {
-			...generateAnalyticsDateRangeArgs( analyticsRangeDates ),
-			'_r.drilldown': analyticsDrilldownArgs.join( ',' ),
-		} )
+				case 'bounce-rate':
+					return getServiceReportURL( 'visitors-overview', {
+						...generateAnalyticsDateRangeArgs( rangeDates ),
+						'_r.drilldown': drilldownArgs.join( ',' ),
+					} );
+			}
```

Comment:
Nit-pick: There are some duplicate function calls of similar nature in each of these `switch` cases, so we could alternatively use that `switch` only to define the values for the first `getServiceReportURL` parameter plus the drilldown arguments to use.

Reply:
I looked at splitting out common args but I didn't feel like it offered much benefit in exchange for further splitting up the code.  I think having a few duplicate calls is fine in this case, especially in a switch where only one path will run. It's a bit safer to keep it isolated by case as well.